### PR TITLE
[AIRFLOW-2307] Add uuid attribute to TaskInstance, return uuid in Executor

### PR DESCRIPTION
**Pull request by johnarnold**
…k id

Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2307
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Closes AIRFLOW-2307.  Adds uid to task instance, opportunistically set uid if an id is returned from execute_async.  Returns task id from CeleryExecutor.   This should be generally applicable to other executors with a little work.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Covered by existing tests.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by bolkedebruin** [comment]
do you mean UUID?

**Comment by johnarnold** [comment]
@bolkedebruin I don't have any religion about uuid/guid, simplified as uid out of habit.  I'll change it to uuid to be consistent with convention.

**Comment by johnarnold** [comment]
Updated to uuid.  Thanks @bolkedebruin for the feedback.

File: airflow/migrations/versions/d02246ccf39b_add_uuid_field_to_task_instance.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+down_revision = '0e2a74e0fc9f'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.add_column('task_instance', sa.Column('uyid', sa.String(length=250), nullable=True))
```
Comment (msumit) [inline airflow/migrations/versions/d02246ccf39b_add_uuid_field_to_task_instance.py]: typo? `uyid`

Comment (johnarnold) [reply airflow/migrations/versions/d02246ccf39b_add_uuid_field_to_task_instance.py]: @msumit good eye!  Fixed.

File: airflow/executors/base_executor.py [hunk 2]
```diff
@@ -128,7 +128,9 @@ def heartbeat(self):
             ti.refresh_from_db()
             if ti.state != State.RUNNING:
                 self.running[key] = command
-                self.execute_async(key, command=command, queue=queue)
+                uuid = self.execute_async(key, command=command, queue=queue)
+                if uuid:
```
Comment (Fokko) [inline airflow/executors/base_executor.py:140]: I like the idea of keeping track of the `uuid`'s. Just curious, are there any situations when you get a null?

**Comment by bolkedebruin** [comment]
I do think this needs to be more comprehensive, ie. support the other executors as well.

**Comment by bolkedebruin** [comment]
and it needs tests, as with the typos that were in it would have failed in any production env.

File: airflow/executors/base_executor.py (hunk 2 above)
Comment (johnarnold) [reply airflow/executors/base_executor.py:140]: @Fokko if the executor doesn't return anything, The uuid will be a None, and shows up empty in the webserver column etc.

**Comment by johnarnold** [comment]
@bolkedebruin It should be pretty easy to support uuids in all the executors, but I made this in such a way that it's totally optional.  I don't use the other executors generally, but I can take a look at the unit tests and see if this is something I can make time to do.  Certainly LocalExecutor and SequentialExecutor should be easy.  Otherwise, I figured someone who uses the alternatives (dask, etc) would know the best way to do it.

It's my bad for changing uid -> uuid and not actually re-running the unit tests before committing :|   I was going to just confirm it worked in travic-ci.   What specific tests would you like to see?

I am running this in production.  I was running the code before uid -> uuid change. Now I'm running the latest commit.

**Comment by johnarnold** [comment]
Updated executors to return uuid (or Future.key for dask, which appears to be a prefixed uuid).  Working on tests.

**Comment by johnarnold** [comment]
Added a couple simple tests for the executors returning uuid.  Not sure how to test that the uuid gets set in the model.

**Comment by johnarnold** [comment]
Rebased (attempting to get CI builds to pass)

**Comment by ron819** [comment]
@johnarnold  are you still working on this PR?

**Pull request closed without merging**